app/javascript/components/index-table/components/data-table.jsx

```diff
@@ -136,6 +139,51 @@ const Datatable = ({
     }
   };
 
+  function sortByProperty(arr, prop) {
```

Comment:
Not much or nothing in this file should have changed. I believe the issue is somewhere around the find method of this file (app/javascript/db/collections/records.js). Maybe the metadata isn't being pass to it? Data is sorted/filtered before this component uses it. That way this component should only have to pretty much render it. It also stops us from hard coding field names in this component to figure out what to sort on using the metadata that is passed from the table. Sorting was working in a previous version of Primero, I suspect it broke somewhere related to the file mentioned above.

Reply:
As sorting is implemented from the API, so for offline mode, we can't access DB collections to retrieve data for sorting. Because of that, implemented a custom method to sort the records in offline mode.